pkg/runtime/core/trainingruntime.go

```diff
@@ -187,24 +187,46 @@ func (r *TrainingRuntime) mergePodSpecOverrides(trainJob *trainer.TrainJob, jobS
 			}) {
 				continue
 			}
-			patch, err := json.Marshal(podSpecOverride)
+
+			overrideBytes, err := json.Marshal(podSpecOverride)
 			if err != nil {
 				return err
 			}
-			source, err := json.Marshal(job.Template.Spec.Template.Spec)
+			overrideMap := map[string]any{}
+			if err := json.Unmarshal(overrideBytes, &overrideMap); err != nil {
+				return err
+			}
+			podTemplatePatch := map[string]any{}
+			md := map[string]any{}
+			if labels, ok := overrideMap["labels"]; ok {
+				md["labels"] = labels
+			}
+			if annotations, ok := overrideMap["annotations"]; ok {
+				md["annotations"] = annotations
+			}
+			podTemplatePatch["metadata"] = md
+			// Note: labels and annotations remain in overrideMap, but strategic merge patch
+			// will ignore them in the spec section as they're not valid PodSpec fields
+			podTemplatePatch["spec"] = overrideMap
```

Comment:
Did you try to keep `.spec` values in `.metadata` as well? I think, values should also be ignored.
```suggestion
			overrideMap := map[string]any{}
			if err := json.Unmarshal(overrideBytes, &overrideMap); err != nil {
				return err
			}
			podTemplatePatch := map[string]any{}
			podTemplatePatch["metadata"] = overrideMap
			// Note: labels and annotations remain in overrideMap, but strategic merge patch
			// will ignore them in the spec section as they're not valid PodSpec fields
			podTemplatePatch["spec"] = overrideMap
```

Reply:
Updated.